[doc] maintain document about sklearn (#27081)
ray-project/ray: doc/source/ray-more-libs/joblib.rst

.. _`issue on GitHub`: https://github.com/ray-project/ray/issues

Ray supports running distributed `scikit-learn`_ programs by
implementing a Ray backend for `joblib`_ using `Ray Actors <../ray-core/actors.html>`__

> instead of local processes. This makes it easy to distribute existing application process that use scikit-learn from a single node to a cluster.

Reviewer: I prefer the existing wording here, I think talking about processes is too low level for this description. if we want to add it, we can add a more detailed "architecture" section

Quickstart
----------

> `Install Ray <../ray-overview/installation.html>`__ and call some methods in your scikit-learn code as the following.

Reviewer: ```suggestion
`Install Ray <../ray-overview/installation.html>`__ and call some methods in your scikit-learn code as follows.
```

.. code-block:: python

  import joblib
  from ray.util.joblib import register_ray
  register_ray()
  with joblib.parallel_backend('ray'):
      # scikit-learn code

Usage
-----

> ``from ray.util.joblib import register_ray`` imports joblib with Ray backend library.

Reviewer: ```suggestion
``from ray.util.joblib import register_ray`` imports the Ray backend for joblib.
```

> ``register_ray()`` registers Ray as a joblib backend for scikit-learn to use.

Reviewer: ```suggestion
``register_ray()`` registers Ray as the joblib backend for scikit-learn to use.
```

Reviewer: also, you can use backticks to format the library names like: `joblib` and `scikit-learn`

> The program inside ``with joblib.parallel_backend()`` is executed on the backend used by Parallel, and 

Reviewer: ```suggestion
The program inside ``with joblib.parallel_backend()`` is executed using the default backend registered for parallel execution and 
```

> the program inside ``with joblib.parallel_backend('ray')`` is executed on Ray as the backend used by Parallel.
> Then, ``with joblib.parallel_backend('ray')`` deploys your code on a Ray cluster. Inside the call, run your original scikit-learn code.

Reviewer: ```suggestion
the program inside ``with joblib.parallel_backend('ray')`` is executed on Ray as the backend used by Parallel.
Then, ``with joblib.parallel_backend('ray')`` deploys your code on a Ray cluster. Inside the call, run your original scikit-learn code.
```

```suggestion
the program inside ``with joblib.parallel_backend('ray')`` is executed using the Ray backend for parallel execution. Your `scikit-learn` code will be distributed to actors across the cluster and executed in parallel.
```

  import joblib
  from ray.util.joblib import register_ray
  register_ray()

>   # ray.init(address=<address>) # if necessary

Reviewer: you can drop this, it's implicitly called

  with joblib.parallel_backend('ray'):
      search.fit(digits.data, digits.target)

> This will start a local Ray cluster. 

Reviewer: ```suggestion
By default this will start a local Ray cluster. 
```
